src/main/java/com/manywho/services/sharepoint/client/HttpClient.java

```diff
@@ -26,11 +26,27 @@ public String executeRequest(HttpRequestBase requestBase){
 
                 if (status >= 200 && status < 300) {
                     HttpEntity entity = httpResponse.getEntity();
-
                     return EntityUtils.toString(entity);
                 }
 
-                throw new RuntimeException(httpResponse.getStatusLine().toString());
+                String errorLine = httpResponse.getStatusLine().toString();
+
+                // If we get an error about the admin consent, we send a better error description
+                if (httpResponse.getEntity() != null) {
+                    try {
+                        String entityError = EntityUtils.toString(httpResponse.getEntity());
+                        JSONObject objectError = new JSONObject(entityError);
+                        String errorDescription = objectError.getString("error_description");
+
+                        if (errorDescription.contains("AADSTS65001")) {
+                            errorLine = "The user or administrator has not consented to use this application." +
```

Comment:
I'm going to guess someone using the Sharepoint service will understand what "Send an interactive authorization request for this user and resource" means, as I don't :wink:

Reply:
@jonjomckay yes, it is what it says in the docs, so I just keep it very similar.